stdlib/LinearAlgebra/src/hessenberg.jl

```diff
@@ -90,6 +90,10 @@ Base.isassigned(H::UpperHessenberg, i::Int, j::Int) =
 Base.@propagate_inbounds getindex(H::UpperHessenberg{T}, i::Integer, j::Integer) where {T} =
     i <= j+1 ? convert(T, H.data[i,j]) : zero(T)
 
+Base._reverse(A::UpperHessenberg, dims) = reverse(Matrix(A); dims)
+Base._reverse(A::UpperHessenberg, dims::Colon) = (reverse(Matrix(A)), A.data)
+Base._reverse!(A::UpperHessenberg, dims::Colon) = (reverse!(A.data); A)
```

Comment:
```suggestion
```

This should still throw. It is not generally possible to reverse an UpperHessenberg matrix because the top right entries should move to the bottom left which are structural zeros.

```julia
julia> uh = UpperHessenberg(rand(1:100, 5, 5))
5×5 UpperHessenberg{Int64, Matrix{Int64}}:
 15  90  87  98  75
 47  73  24  97   9
  ⋅  94  15  34  97
  ⋅   ⋅  97  39  67
  ⋅   ⋅   ⋅  64  88

julia> reverse!(uh.data); uh
5×5 UpperHessenberg{Int64, Matrix{Int64}}:
 88  64   7  67  29
 67  39  97  49  89
  ⋅  34  15  94   7
  ⋅   ⋅  24  73  47
  ⋅   ⋅   ⋅  90  15
```